supabase/client.py

```diff
@@ -200,9 +202,11 @@ def _init_postgrest_client(
     def _get_auth_headers(self) -> Dict[str, str]:
         """Helper method to get auth headers."""
         # What's the corresponding method to get the token
+        data = self.auth.get_session()
+        token = data.get("access_token") if data else None
         return {
             "apiKey": self.supabase_key,
-            "Authorization": f"Bearer {self.supabase_key}",
+            "Authorization": f"Bearer {token}",
```

Comment:
I think this line is executed only when the client is initialized, that is before sign-in. We need something that sets the header not only after sign-in but also every time the access token is refreshed. Maybe we can add a function like `Client.ensure_fresh_session()` that would get the token from `auth.get_session()` (which refreshes access token if needed) and then set the authorization header for all sub-components of the `Client`.

Reply:
Yup, I agree, there's a refresh that is needed. In the js lib there's a refresh triggered when the auth state changes. I don't quite have bandwidth to look into this outside of weekends but if anyone has bandwidth feel free to pick it up

Reply:
It feels natural for the correct token to be set when calling `Client..auth.set_session()`. Could the headers just be updated as a part of this workflow?

Reply:
Then the typical workflow will be:
1. sign_in
2. set_session()
3. call to retrieve data from the DB
// some time later, when the access token might have expired
4. set_session() again
5. another call to retrieve data from the DB

Forcing people to call `set_session()` every time before DB calls seems redundant. Am I missing something?

We could change the implementation of `Client.from_()` to take care of token refresh and header, then the user can work in a more intuitive way. 
1. sign in
2. call DB